sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/IntervalUtilsSuite.scala

```diff
@@ -379,6 +384,15 @@ class IntervalUtilsSuite extends SparkFunSuite with SQLHelper {
     assert(negate(new CalendarInterval(1, 2, 3)) === new CalendarInterval(-1, -2, -3))
   }
 
+  test("parsing second_nano string") {
+    val parsingException = intercept[SparkIllegalArgumentException] {
+      toDTInterval("12", "33.33.33", 1)
+    }
+
+    assert(parsingException.getErrorClass === "INTERVAL_ERROR.SECOND_NANO_FORMAT")
+    assert(parsingException.getSqlState === "22009")
+  }
+
```

Comment:
Please, use `checkError()`.

Reply:
I tried using `checkError()`, however, it is convenient for `SparkRuntime` exceptions. All of the exceptions thrown in this case are `SparkIllegalArgument` exceptions, thus the parameters are not adequate for checking with the function `checkError()`. If you want, I can convert this one and maybe some others to `SparkRuntime` exceptions and then check them with `checkError()`

Reply:
>  thus the parameters are not adequate for checking with the function checkError()

Please, explain this.

> If you want, I can convert this one

Please, change this to:
```scala
    checkError(
      exception = intercept[SparkIllegalArgumentException] {
        toDTInterval("12", "33.33.33", 1)
      },
      errorClass = "INTERVAL_ERROR.SECOND_NANO_FORMAT",
      sqlState = Some("22009")
    )
```